File: sources/python/05_alloc_churn.py
# task 05 alloc_churn — expected output: 1274991808
# build: python 05_alloc_churn.py | pypy 05_alloc_churn.py | graalpy 05_alloc_churn.py | nuitka --standalone 05_alloc_churn.py    run: python 05_alloc_churn.py

total = 0
slots = [None] * 256

for i in range(10000000):
    buf = bytearray(64)
    buf[0] = i % 256
    total += buf[0]
    slots[i % 256] = buf

print(total)

File: sources/go/05_alloc_churn.go
// task 05 alloc_churn — expected output: 1274991808
// build: go build -o prog 05_alloc_churn.go    run: ./prog
// note: plain "go build" only; do not set GOGC=off or any other tuning flags
package main

import "fmt"

func main() {
	var slots [256][]byte
	var total uint64
	for i := 0; i < 10000000; i++ {
		buf := make([]byte, 64)
		buf[0] = byte(i % 256)
		total += uint64(buf[0])
		slots[i%256] = buf
	}
	fmt.Println(total)
}
